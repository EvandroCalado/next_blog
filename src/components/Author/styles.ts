import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
