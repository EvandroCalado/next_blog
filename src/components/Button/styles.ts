import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
