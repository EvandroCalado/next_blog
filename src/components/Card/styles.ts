import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    a {
      transition: all 300ms ease-in-out;
    }

    a:hover {
      opacity: 1;
      filter: brightness(170%);
    }
`;

export const Cover = styled.div`
  ${({ theme }) => css`
    img {
      width: 100%;
      display: block;
      aspect-ratio: 16/9;
      border-radius: .5rem;
      margin-bottom: ${theme.spacings.small};

    }
  `}
`;

export const Heading = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.smallMedium};
    color: ${theme.colors.darkGray};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`;

export const Paragraph = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.primary};
  `}
`;
