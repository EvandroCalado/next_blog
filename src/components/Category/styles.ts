import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 2rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  width: 8rem;
  border-radius: 0.3rem;
`;

export const Title = styled.h2`
${({ theme }) => css`
  font-size: ${theme.font.sizes.xlarge};

  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.large};
  }
`}
`;
