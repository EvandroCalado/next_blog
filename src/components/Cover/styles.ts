import styled, { css } from 'styled-components';

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    aspect-ratio: 16/9;
    object-fit: cover;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
