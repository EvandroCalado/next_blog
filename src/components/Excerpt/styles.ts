import styled, { css } from 'styled-components';

export const Paragraph = styled.p`
  ${({ theme }) => css`
    width: 100%;
    margin: ${theme.spacings.xxlarge} 0;
  `}
`;
