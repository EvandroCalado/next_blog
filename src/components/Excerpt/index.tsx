import React from 'react';
import * as Styled from './styles';

export type ExcerptProps = {
  excerpt: string;
};

const Excerpt = ({ excerpt }: ExcerptProps) => {
  return (
    <Styled.Paragraph data-testid="excerpt-wrapper">{excerpt}</Styled.Paragraph>
  );
};

export default React.memo(Excerpt);
