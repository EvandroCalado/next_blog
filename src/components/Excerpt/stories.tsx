import Excerpt, { ExcerptProps } from './';
import { Meta, StoryFn } from '@storybook/react';

export default {
  title: 'Excerpt',
  component: Excerpt,
  args: {
    excerpt:
      'Lorem markdownum urit, prohibete posuere annum suos fecimus, per, Lorem markdownum urit, prohibete posuere annum suos fecimus, per',
  },
} as Meta;

export const Template: StoryFn<ExcerptProps> = (args) => (
  <div>
    <Excerpt {...args} />
  </div>
);
