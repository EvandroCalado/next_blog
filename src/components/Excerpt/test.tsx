import Excerpt from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

describe('<Excerpt />', () => {
  it('test_excerpt_rendered_with_valid_prop', () => {
    const excerpt = 'This is a valid excerpt';

    renderTheme(<Excerpt excerpt={excerpt} />);
    expect(screen.getByText(excerpt)).toBeInTheDocument();
  });

  it('test_rendering_empty_excerpt', () => {
    const excerpt = '';

    renderTheme(<Excerpt excerpt={excerpt} />);
    expect(screen.getByTestId('excerpt-wrapper')).toBeInTheDocument();
    expect(screen.getByTestId('excerpt-wrapper')).toHaveTextContent(excerpt);
  });

  it('should render a snapshot', () => {
    const excerpt = 'This is a valid excerpt';

    const { container } = renderTheme(<Excerpt excerpt={excerpt} />);
    expect(container).toMatchSnapshot();
  });
});
