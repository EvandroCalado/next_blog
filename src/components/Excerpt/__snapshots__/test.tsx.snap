// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Excerpt /> should render a snapshot 1`] = `
.c0 {
  width: 100%;
  margin: 4.8rem 0;
  text-align: center;
}

<div>
  <p
    class="c0"
    data-testid="excerpt-wrapper"
  >
    This is a valid excerpt
  </p>
</div>
`;
